src/BbsBlsSignatureProof2020.ts

```diff
@@ -252,14 +254,12 @@ export class BbsBlsSignatureProof2020 extends suites.LinkedDataProof {
       // Transform the blank node identifier placeholders for the document statements
       // back into actual blank node identifiers
       const transformedDocumentStatements = documentStatements.map(element => {
-        const nodeIdentifier = element.split(" ")[0];
-        if (nodeIdentifier.startsWith("<urn:bnid:_:c14n")) {
+        if (element.includes("<urn:bnid:_:c14n")) {
+          const prefixIndex = element.indexOf("<urn:bnid:");
+          const closingIndex = element.indexOf(">", prefixIndex);
           return element.replace(
-            nodeIdentifier,
-            nodeIdentifier.substring(
-              "<urn:bnid:".length,
-              nodeIdentifier.length - 1
-            )
+            element.substring(prefixIndex, closingIndex + 1),
+            element.substring(prefixIndex + 10, closingIndex)
```

Comment:
Hmm, why `+ 10` here?

Reply:
I'm glad you caught this cause I didn't have a better solution `<urn:bnid:` is 10 characters it's definitely magic string style so I can add some comments in here if we leave this solution

Reply:
```suggestion
            element.substring(prefixIndex + "<urn:bnid".length(), closingIndex)
```